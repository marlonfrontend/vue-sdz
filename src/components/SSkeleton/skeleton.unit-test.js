import { mount } from '@vue/test-utils'

import Skeleton from './Index.vue'

/* eslint-disable */
describe('Skeleton', () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(Skeleton, {
      propsData:{
        size: 65,
        rows: 3
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check class', () => {

    const cl = cmp.find('div')

    expect(cl.classes()).toContain('s-skeleton-container')
  })

  test('check style', () => {

    const cl = cmp.find('div > div')

    expect(cl.attributes().style).toContain('width: 65%; height: 60px;')
  })

})
