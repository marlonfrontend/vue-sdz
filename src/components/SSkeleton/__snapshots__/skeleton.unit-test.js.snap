// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton has the expected html structure 1`] = `
<div
  class="s-skeleton-container"
>
  <div
    class="s-skeleton"
    style="width: 65%; height: 60px;"
  >
    <div
      class="skeleton-background"
      style="height: 60px; border-radius: $border-radius-sm;"
    />
  </div>
</div>
`;
